docs/reqcontext.rst

```diff
@@ -84,12 +84,37 @@ The request context internally works like a stack: The topmost level on
 the stack is the current active request.
 :meth:`~flask.ctx.RequestContext.push` adds the context to the stack on
 the very top, :meth:`~flask.ctx.RequestContext.pop` removes it from the
-stack again.  On popping the application's
+stack again.  On popping, the application's
 :func:`~flask.Flask.teardown_request` functions are also executed.
 
-Another thing of note is that the request context will automatically also
-create an :ref:`application context <app-context>` when it's pushed and
-there is no application context for that application so far.
+.. _request-context-and-app-context:
+
+How the Request Context and Application Context Work Together
+-------------------------------------------------------------
+
+Normally, when a request context is pushed, the request context pushes an
+application context as well.  Then, when the request context is popped, it
+pops the application context as well, which executes the
+:func:`~flask.Flask.teardown_appcontext` functions.
+
+However, if a request context is pushed and there's already an application
+context (for the request context's app) on the stack, the request context
+won't* push a new app context but will reuse the existing one.  The request
```

Comment:
Shouldn't be there, at least not without one before it in case it should be italic..